src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs

```diff
@@ -352,6 +372,14 @@ public override byte[] ComputeJwkThumbprint()
                 throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX10706, nameof(Kty), string.Join(", ", JsonWebAlgorithmsKeyTypes.EllipticCurve, JsonWebAlgorithmsKeyTypes.RSA, JsonWebAlgorithmsKeyTypes.Octet), nameof(Kty))));
         }
 
+        private bool CanComputeOctThumbprint()
+        {
+            if (string.IsNullOrEmpty(K))
```

Comment:
could we return: !string.IsNullOrEmpty(K);

Reply:
I'll remove all unnecessary if-branches.

Reply:
Addressed with https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1346/commits/303138374134fe6c096ded0ade0b448252f8f695.